### PR TITLE
CIV-5853 mediation navigation

### PR DESCRIPTION
**Pull request by GorilaNaranja**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5853

### Change description ###
Mediation Navigation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/services/features/response/mediation/companyTelephoneNumberService.ts [hunk 1]
```diff
@@ -19,15 +20,29 @@ const filterFormWithSelection = (form: CompanyTelephoneNumber) => {
...
-    const contactPerson = claim.respondent1?.partyDetails.contactPerson;
-    let telephoneNumberData = new CompanyTelephoneNumber();
 
-    if (claim.mediation?.companyTelephoneNumber) {
-      telephoneNumberData = claim.mediation.companyTelephoneNumber;
-    } else if (claim.respondent1?.partyPhone) {
-      telephoneNumberData.mediationPhoneNumberConfirmation = claim.respondent1?.partyPhone?.phone;
+    if (claim.isClaimantIntentionPending()) {
+      const contactPerson = claim.applicant1?.partyDetails?.contactPerson;
+      let telephoneNumberData = new CompanyTelephoneNumber();
```
Comment (laperla11) [inline src/main/services/features/response/mediation/companyTelephoneNumberService.ts:26]: You can remove this line before 24 and delete line 31

**Review by laperla11 — DISMISSED** [review]

File: src/main/services/features/response/mediation/companyTelephoneNumberService.ts (hunk 1 above)
Comment (GorilaNaranja) [reply src/main/services/features/response/mediation/companyTelephoneNumberService.ts:26]: done

File: src/main/services/features/response/mediation/companyTelephoneNumberService.ts [hunk 2]
```diff
@@ -19,15 +20,28 @@ const filterFormWithSelection = (form: CompanyTelephoneNumber) => {
...
   try {
     const claim = await getCaseDataFromStore(claimId);
-    const contactPerson = claim.respondent1?.partyDetails.contactPerson;
     let telephoneNumberData = new CompanyTelephoneNumber();
 
-    if (claim.mediation?.companyTelephoneNumber) {
-      telephoneNumberData = claim.mediation.companyTelephoneNumber;
-    } else if (claim.respondent1?.partyPhone) {
-      telephoneNumberData.mediationPhoneNumberConfirmation = claim.respondent1?.partyPhone?.phone;
+    if (claim.isClaimantIntentionPending()) {
```
Comment (LeonardoPalmeiro) [inline src/main/services/features/response/mediation/companyTelephoneNumberService.ts:25]: we can substitute this part for const contactPerson = claim.isClaimantIntentionPending()
  ? claim.applicant1?.partyDetails?.contactPerson
  : claim.respondent1?.partyDetails.contactPerson;
  what do you think?

Comment (LeonardoPalmeiro) [reply src/main/services/features/response/mediation/companyTelephoneNumberService.ts:25]: even better separate into two different methods one for claimant for respondent then this part of the code will be more readable.

Comment (GorilaNaranja) [reply src/main/services/features/response/mediation/companyTelephoneNumberService.ts:29]: I extracted the contactPerson variable as you suggested. The other suggestion needs a bigger refactor.

**Review by laperla11 — APPROVED** [review]

**Pull request merged**